src/client/components/input/partial-date.jsx

```diff
@@ -25,102 +25,91 @@ const ymdRegex = /^\d{4}-\d{2}-\d{2}$/;
 const ymRegex = /^\d{4}-\d{2}$/;
 const yRegex = /^\d{4}$/;
 
-(() => {
-	'use strict';
-
-	class PartialDate extends React.Component {
-		static validate(value) {
-			if (!value) {
-				return true;
-			}
-
-			const validSyntax = Boolean(
-				ymdRegex.test(value) ||
-				ymRegex.test(value) ||
-				yRegex.test(value)
-			);
-			const validValue = !isNaN(Date.parse(value));
-
-			return validSyntax && validValue;
-		}
-
-		constructor(props) {
-			super(props);
-
-			this.state = {
-				value: this.props.defaultValue,
-				valid: PartialDate.validate(this.props.defaultValue)
-			};
-
-			// React does not autobind non-React class methods
-			this.handleChange = this.handleChange.bind(this);
-		}
-
-		getValue() {
-			return this.validate(this.input.getValue().trim());
```

Comment:
This line is the problem. However once this is changed this component returns true/false instead of the date, which causes problems in other areas